### PR TITLE
Converge main and next branch

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR fast-forwards main to next. It needs to be opened so that GitHub runs the necessary checks to allow the fast-forward.

### CODE REVIEW

**Review by CraigMacomber — APPROVED** [review]

File: experimental/dds/tree2/package.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
 	"module": "lib/index.js",
 	"types": "dist/index.d.ts",
 	"scripts": {
-		"bench": "mocha --timeout 999999 --perfMode --parentProcess --fgrep @Benchmark --reporter @fluid-tools/benchmark/dist/MochaReporter.js",
-		"bench:profile": "mocha --v8-prof --timeout 999999 --perfMode --fgrep @Benchmark --reporter @fluid-tools/benchmark/dist/MochaReporter.js && node --prof-process isolate-0x*-v8.log > profile.txt && rm isolate-0x*-v8.log && cat profile.txt",
+		"bench": "mocha --timeout 999999 --perfMode --parentProcess --fgrep @Benchmark --reporter \"../../../node_modules/@fluid-tools/benchmark/dist/MochaReporter.js\"",
```
Comment (CraigMacomber) [inline experimental/dds/tree2/package.json:18]: I think this is reverting a fix I did.

File: experimental/dds/tree2/package.json [hunk 2]
```diff
@@ -37,7 +37,7 @@
 		"prettier": "prettier --check . --ignore-path ../../../.prettierignore",
 		"prettier:fix": "prettier --write . --ignore-path ../../../.prettierignore",
 		"test": "npm run test:mocha",
-		"test:benchmark:report": "mocha --exit --perfMode --parentProcess --fgrep @Benchmark --reporter @fluid-tools/benchmark/dist/MochaReporter.js --timeout 60000",
+		"test:benchmark:report": "mocha --node-option unhandled-rejections=strict,expose-gc --exit --perfMode --parentProcess  --fgrep @Benchmark --reporter @fluid-tools/benchmark/dist/MochaReporter.js --timeout 60000",
```
Comment (CraigMacomber) [inline experimental/dds/tree2/package.json:40]: I think these options come from the moch config and were removed on purpose, so I think this is also a regression. Perhaps a bad merge?

File: .vscode/settings.json [hunk 3]
```diff
@@ -25,19 +25,23 @@
 	// Custom dictionary for 'streetsidesoftware.code-spell-checker' extension.
 	"cSpell.words": [
 		"endregion",
+		"handlecache",
```
Comment (CraigMacomber) [inline .vscode/settings.json:28]: a lot of the added words here are not words. I'm not sure adding them is a good idea.

File: package.json [hunk 4]
```diff
@@ -54,10 +54,8 @@
 		"lint": "npm run syncpack:deps && npm run syncpack:versions && npm run prettier:root && npm run ci:eslint",
 		"lint:fix": "npm run prettier:root:fix && pnpm run -r --no-sort --stream lint:fix",
 		"policy-check": "flub check policy --exclusions build-tools/packages/build-tools/data/exclusions.json",
-		"policy-check-help": "echo Detect (and error on) policy-check violations, like package.json sorting, copyright headers etc. Excludes assert-short-code. Run the check or \"pnpm flub check policy --listHandlers\" for a full list.",
```
Comment (CraigMacomber) [inline package.json:57]: why was this deleted. I wrote it for a reason. Did we have a bunch of merges just losing changes to package.json files when version updates happened?

**Comment by CraigMacomber** [comment]
Its hard to view on github, but the top level package lock downgrades a bunch of package versions. Not ideal.

**Comment by CraigMacomber** [comment]
I approved this as this isn't really the place to review these changes, they were already reviewed going into next, but just looking at a few places of interest to me where I did changes, I can see regression where work I did has been lost. I don't know how many other places are impacted as well, and we might want to do more detailed review somehow. I'm not sure what should be done.

File: package.json (hunk 4 above)
Comment (tylerbutler) [reply package.json:57]: This is very concerning. I think you're probably right that this was due to a merge resolution gone wrong. But how to find them and prevent them? Not sure.

File: common/lib/common-definitions/pnpm-lock.yaml [hunk 5]
```diff
@@ -108,18 +104,18 @@ packages:
       jsdoc-type-pratt-parser: 3.1.0
     dev: true
 
-  /@eslint-community/eslint-utils@4.4.0(eslint@8.40.0):
+  /@eslint-community/eslint-utils@4.4.0(eslint@8.39.0):
```
Comment (CraigMacomber) [inline common/lib/common-definitions/pnpm-lock.yaml:107]: bunch of package downgrades here as well (just like the root package lock)

File: examples/utils/bundle-size-tests/package.json [hunk 6]
```diff
@@ -43,7 +43,7 @@
 	"devDependencies": {
 		"@cerner/duplicate-package-checker-webpack-plugin": "^2.3.0",
 		"@fluidframework/build-common": "^1.1.0",
-		"@fluidframework/bundle-size-tools": "^0.17.0",
+		"@fluidframework/bundle-size-tools": "^0.15.0",
```
Comment (CraigMacomber) [inline examples/utils/bundle-size-tests/package.json:46]: this downgrade seems bad or at least unintended.

**Comment by tylerbutler** [comment]
> Its hard to view on github, but the top level package lock downgrades a bunch of package versions. Not ideal.

Ugh, I probably did this. I think I merged the wrong direction on the lock file - I think I took the bad stuff from next and used that as the base, effectively undoing your work that I merged. 😦

File: BREAKING.md [hunk 7]
```diff
@@ -15,9 +15,121 @@ It's important to communicate breaking changes to our stakeholders. To write a g
...
+`IProvideContainerRuntime` and its `IContainerRuntime` member were deprecated in a previous release and have been removed. This applies to the `ContainerRuntime` class as well.
+
+### IntervalCollection removed
+
+The exports deprecated in [IntervalCollection public export deprecated](#intervalCollection-public-export-deprecated) have been removed.
+
 # 2.0.0-internal.4.4.0
 
-## 2.0.0-internal.4.3.0 Upcoming changes
+## 2.0.0-internal.4.4.0 Upcoming changes
```
Comment (CraigMacomber) [inline BREAKING.md:132]: Given that this branch is on version internal.5 now, .4.4.0 changes shouldn't be listed as upcoming.

**Comment by tylerbutler** [comment]
> I approved this as this isn't really the place to review these changes, they were already reviewed going into next, but just looking at a few places of interest to me where I did changes, I can see regression where work I did has been lost. I don't know how many other places are impacted as well, and we might want to do more detailed review somehow. I'm not sure what should be done.

I'll merge to get the branches unlocked, but high pri tomorrow will be unraveling what happened and undoing any potential damage before we release 5.0. I already created the release branch for 4.4, so it's ready to go.

File: examples/benchmarks/bubblebench/editable-shared-tree/package.json [hunk 8]
```diff
@@ -39,6 +36,8 @@
 		"test:jest": "jest",
 		"test:jest:verbose": "cross-env FLUID_TEST_VERBOSE=1 jest",
 		"tsc": "tsc",
+		"tsfmt": "tsfmt --verify",
```
Comment (CraigMacomber) [inline examples/benchmarks/bubblebench/editable-shared-tree/package.json:39]: these lines were removed in https://github.com/microsoft/FluidFramework/commit/2e295e0a6be407f773cf4b3b756f13127dd00c34#diff-3c675c6b50addad660dbb73bc376ae3b0b39383a2a9c54ca101d8806b196644d . This loses that cleanup.
